pkg/controller/che/k8s_helpers.go

```diff
@@ -267,13 +267,16 @@ func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, url string)
 			return nil, err
 		}
 		// sometimes timing conditions apply, and host isn't available right away
-		if len(testRoute.Spec.Host) < 1 {
+		for i := 0; i < 10 && len(testRoute.Spec.Host) < 1; i++ {
```

Comment:
I'm not a big fan of increasing the timeout from 1 second to 10 seconds. Afaik normally a operator controller should be as less blocking as possible.
I assume in our case it is acceptable until some wider refactoring is done.
However, since you return an error now when the route is not found, it might be interesting to test if the operator wouldn't simply trigger this request again after the error has been return, so that the process would resume without blocking the request loop.

Reply:
For me, it works even from the first try, I do not have an idea why testRoute.Spec.Host can be empty here, maybe it can be reproduced on some slow infrastructure...

So, timeout is set to 3 seconds to make operator less blocking.

> since you return an error now when the route is not found

Sorry but it's not an error when route is not found, but when route is found without host... And I'm not able to reproduce it on my local installation.

BTW Thanks that you caught my attention at that error, I figure it out that we must delete test route before returning an error.

Reply:
@davidfestal cc

Reply:
> However, since you return an error now when the route is not found, i

Let me clarify that I return an error if route is found but without host.

I've rewritten it in non-blocking way but I'm able to test only positive scenario when the route has host just after creation.
I do not have an idea why testRoute.Spec.Host can be empty here, maybe it can be reproduced on some slow infrastructure.

Reply:
The new approach is tested by simulating a case when test route does not have host immediately
```go
func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, url string) (certificate []byte, err error) {
	var requestURL string
	var testRoute *routev1.Route
	if len(url) < 1 {
		testRoute = r.GetEffectiveRoute(instance, "test")
		if testRoute == nil {
			testRoute = deploy.NewTlsRoute(instance, "test", "test", 8080)
			logrus.Infof("Creating a test route %s to extract router crt", testRoute.Name)
			if err := r.CreateNewRoute(instance, testRoute); err != nil {
				logrus.Errorf("Failed to create test route %s: %s", testRoute.Name, err)
				return nil, err
			}

                        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
			return nil, errors.New("Fail the first time")
		}

		if len(testRoute.Spec.Host) < 1 {
			return nil, errors.New("Unable to get test route host for fetching certificate")
		}
 ...
```
And it seems to work as expected:
![Screenshot_20190904_155704](https://user-images.githubusercontent.com/5887312/64256723-b002e080-cf2c-11e9-8cd5-d24d1db6143e.png)

@davidfestal, thanks for proposing this way!

Reply:
Seems nice. Just added 2 comments about these changes:
- https://github.com/eclipse/che-operator/pull/66#discussion_r320832304
- https://github.com/eclipse/che-operator/pull/66#discussion_r320836623